simulator/kernel/mars/mars_kernel.cpp

```diff
@@ -228,9 +232,26 @@ void MARSKernel::read_from_file() {
     mem->memcpy_host_to_guest( buffer_ptr, byte_cast( buffer.data()), chars_to_read);
 }
 
-void MARSKernel::connect_memory( std::shared_ptr<FuncMemory> m)
+void MARSKernel::connect_riscv_handler()
+{
+    constexpr Addr TRAP_VECTOR = 0x8'000'0000;
+    sim->write_csr_register( "stvec", TRAP_VECTOR);
+    auto pc = trap_vector_address<Addr>(TRAP_VECTOR, 3);
+    ElfLoader elf_loader( KERNEL_IMAGES "riscv32.bin");
+    elf_loader.load_to( mem.get(), pc - elf_loader.get_text_section_addr());
+}
+
+void MARSKernel::connect_mars_handler()
 {
-    BaseKernel::connect_memory( m);
     ElfLoader elf_loader( KERNEL_IMAGES "mars32_le.bin");
     elf_loader.load_to( mem.get(), 0x8'0000'0180 - elf_loader.get_text_section_addr());
 }
+
+void MARSKernel::connect_exception_handler()
+{
+    auto isa = sim->get_isa();
+    if ( isa == "riscv32")
+        connect_riscv_handler();
+    if ( isa == "mars" || isa == "mips32le" || isa == "mips32")
```

Comment:
Consider using `else if`

Reply:
IIRC we do not support `mips32le` and `mips32`, as kernel is compiled w/o delayed branches,

Reply:
However tests expect that exception handler is connected for `mips32`

Reply:
OK then

Reply:
Could you please clarify what tests?

Reply:
https://github.com/MIPT-ILab/mipt-mips/blob/686f672a61ce77c6394bdfdf53da71c3bb5fea5c/simulator/func_sim/t/unit_test.cpp#L223

https://github.com/MIPT-ILab/mipt-mips/blob/686f672a61ce77c6394bdfdf53da71c3bb5fea5c/simulator/func_sim/t/unit_test.cpp#L210

https://github.com/MIPT-ILab/mipt-mips/blob/686f672a61ce77c6394bdfdf53da71c3bb5fea5c/simulator/func_sim/t/unit_test.cpp#L155